.github/workflows/run-tests.yml

```diff
@@ -21,7 +21,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.8, 3.9, "3.10", 3.11]
+        python-version: ["3.9", "3.10", "3.11"]
```

Comment:
Why the strong drop of 3.8 10 months before Python stops supporting it?

Even if `isx` doesn't support it, you can have conditional requirements that say that `roiextractors` + `isx` support requires `py>3.8`, but the rest of `roiextractors` is cool with it

Reply:
Let's follow [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html), not the [Python release cycle](https://devguide.python.org/versions/). According to NEP 29, Python 3.8 was dropped Apr 14, 2023.

Reply:
We've been following the Python release cycle everywhere else - maybe discuss in the next team meeting?

Reply:
The latest version of numpy, 1.26.0, requires Python 3.9+. If we want to use any new numpy features (or any features that rely on them) we will need to drop support for Python 3.8.

Reply:
What new features would those be, and why would ROIExtractors, NeuroConv, or the GUIDE minimally require them instead of simply letting it be up to the user and their condition installations?

You can also specify `numpy>=1.26.0` conditionally for various Python versions and optional requirements without making the strong decision to retire an entire Python version

Reply:
> This NEP is being put forward by maintainers of Matplotlib, scikit-learn, IPython, Jupyter, yt, SciPy, NumPy, and scikit-image.

Which means that the latest version of all of those libraries is intended to be 3.9+. See for example https://pypi.org/project/scikit-learn/

Reply:
This has been put forth by all of those projects as a best practice for ecosystem support. I don't really want to create any unnecessarily packaging complexity to extend beyond this window

Reply:
@bendichter 
I was not aware of NEP 29. Nice.

I also think that there are minimal costs to dropping support of python 3.8. I think our user base has a lot of freedom over their environments as conda is available on most scientific contexts. As Ben said, I think the cost of increasing packacing complexity is a greater burden for us. If some user comes forward with an issue maybe we can carve out a solution for them but I think it would be easier for us to stick to the scientific python standards.